ttkwidgets/notebook.py

```diff
@@ -0,0 +1,935 @@
+        label = self._tab_labels[tab]
+        x1 = label.winfo_x() / w
+        x2 = x1 + label.winfo_reqwidth() / w
+        xc1, xc2 = self._canvas.xview()
+        if x1 < xc1:
+            self._canvas.xview_moveto(x1)
+        elif x2 > xc2:
+            self._canvas.xview_moveto(xc1 + x2 - xc2)
+        i = self._visible_tabs.index(tab)
+        if i == 0:
+            self._btn_left.state(['disabled'])
+            if len(self._visible_tabs) > 1:
+                self._btn_right.state(['!disabled'])
+        elif i == len(self._visible_tabs) - 1:
+            self._btn_right.state(['disabled'])
+            self._btn_left.state(['!disabled'])
+        else:
+            self._btn_right.state(['!disabled'])
+            self._btn_left.state(['!disabled'])
+
+    def hide(self, tab_id):
+        """Hide tab TAB_ID."""
+        tab = self.index(tab_id)
+        if tab in self._visible_tabs:
+            self._visible_tabs.remove(tab)
+            if tab in self._active_tabs:
+                self._active_tabs.remove(tab)
+            self._hidden_tabs.append(tab)
+            self._tab_labels[tab].grid_remove()
+            if self.current_tab == tab:
+                if self._active_tabs:
+                    self._show(self._active_tabs[0])
+                else:
+                    self.current_tab = -1
+                self._tabs[tab].grid_remove()
+            self.update_idletasks()
+            self._on_configure()
+            self._resize()
+
+    def forget(self, tab_id):
+        """Remove tab TAB_ID from notebook."""
+        tab = self.index(tab_id)
+        if tab in self._hidden_tabs:
+            self._hidden_tabs.remove(tab)
+        elif tab in self._visible_tabs:
+            if tab in self._active_tabs:
+                self._active_tabs.remove(tab)
+            self._visible_tabs.remove(tab)
+            self._tab_labels[tab].grid_forget()
+            if self.current_tab == tab:
+                if self._active_tabs:
+                    self._show(self._active_tabs[0])
+                else:
+                    self.current_tab = -1
+                    if not self._visible_tabs and not self._hidden_tabs:
+                        self._tab_list.state(['disabled'])
+                self._tabs[tab].grid_forget()
+        del self._tab_labels[tab]
+        del self._indexes[str(self._tabs[tab])]
+        del self._tabs[tab]
+        self.update_idletasks()
+        self._on_configure()
+        i = self._tab_menu_entries[tab]
+        for t, ind in self._tab_menu_entries.items():
+            if ind > i:
+                self._tab_menu_entries[t] -= 1
+        self._tab_menu.delete(self._tab_menu_entries[tab])
+        del self._tab_menu_entries[tab]
+        self._resize()
+
+    def move_to_toplevel(self, tab):
+        tl = tk.Toplevel(self)
+        nb = Notebook(tl, **self._init_kwargs)
+        move_widget(tab, nb)
+        nb.add(tab)
+        nb.grid()
+        self._toplevels.append(tl)
+        tl.mainloop()
```

Comment:
`Toplevel.mainloop` need not be called: The `mainloop` of the master window is already running and if it is not, the user of the library is doing something wrong.